tests/vmi_configuration_test.go

```diff
@@ -1415,7 +1415,7 @@ var _ = Describe("[sig-compute]Configurations", func() {
 					Eventually(matcher.ThisVMI(agentVMI), 240*time.Second, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceAgentConnected))
 
 					By("fetching the VMI after agent has connected")
-					Expect(matcher.ThisVMI(agentVMI)()).To(matcher.HaveConditionMissingOrFalse(v1.VirtualMachineInstanceUnsupportedAgent))
+					Eventually(matcher.ThisVMI(agentVMI), 30*time.Second, time.Second).Should(matcher.HaveConditionMissingOrFalse(v1.VirtualMachineInstanceUnsupportedAgent))
```

Comment:
Did you want a `Consistently` here?

Reply:
So there is actually something hidden behind but I am not sure we can do anything about it. Let me explain.

We have the poller that is responsible for caching the GA data.
Now what can happen is that the agent gets connected but we did not yet query the agent or an error occurred while querying the data.
Now the cache is empty and virt-handler gets this empty struct.

[This](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/vm.go#L997) logic will therefore say the agent is not supported for a while. Now I think we would need to distinguish between situations when the cache is empty because we did not perform any attempt and 2case where we did perform it but failed.

I did rather propose this fix for flake rather than solving the real issue because I am not convinced it would make sense to do the effort behind what I described above.

WDYT?

Reply:
I guess it would make sense to to immediately query the guest agent in virt-handler directly following a response which indicates a successful connect.

How often does it fail atm?

Reply:
Your suggestion might be easier to implement, and worth trying.

Ref: https://search.ci.kubevirt.io/?search=AgentVersionNotSupported&maxAge=336h&context=1&type=bug%2Bissue%2Bjunit&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

`162 runs, 10% failed, 12% of failures match = 1% impact`

For some reason, it happens only with one lane...

Reply:
Yeah, let's try it. :+1:  Otherwise we would need at least a FIXME comment and an issue created.

Reply:
Sounds good to me. BTW, I have a weird feeling to be chasing these small impact flakes(outside of what I am used to).

Reply:
It seems we are already doing it. See:
https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go#L236

https://github.com/kubevirt/kubevirt/blob/578603fe064c1da07e676416cc507fbe9be886cb/pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go#L336 
https://github.com/kubevirt/kubevirt/blob/578603fe064c1da07e676416cc507fbe9be886cb/pkg/virt-launcher/notify-client/client.go#L424

Reply:
@rmohr 
Thats my assumption. I saw this one today https://github.com/kubevirt/kubevirt/pull/8997 . I did not check closely if it has an affect but I think so.